ci/github-script/reviewers.js

```diff
@@ -0,0 +1,125 @@
+// Caching the list of team members saves API requests when running the bot on the schedule and
+// processing many PRs at once.
+const members = {}
+
+async function handleReviewers({
+  github,
+  context,
+  core,
+  log,
+  dry,
+  pull_request,
+  reviews,
+  maintainers,
+  owners,
+  getTeamMembers,
+  getUser,
+}) {
+  const pull_number = pull_request.number
+
+  const users = new Set([
+    ...(await Promise.all(
+      maintainers.map(async (id) => (await getUser(id)).login),
+    )),
+    ...owners.filter((handle) => handle && !handle.includes('/')),
+  ])
+  log('reviewers - users', Array.from(users).join(', '))
+  console.log('users', Array.from(users))
+
+  const teams = new Set(
+    owners
+      .map((handle) => handle.split('/'))
+      .filter(([org, slug]) => org === context.repo.owner && slug)
+      .map(([, slug]) => slug),
+  )
+  log('reviewers - teams', Array.from(teams).join(', '))
+
+  const team_members = new Set(
+    (await Promise.all(Array.from(teams, getTeamMembers)))
+      .flat(1)
+      .map(({ login }) => login),
+  )
+  log('reviewers - team_members', Array.from(team_members).join(', '))
+
+  const new_reviewers = users
+    .union(team_members)
+    // We can't request a review from the author.
+    .difference(new Set([pull_request.user?.login]))
+  log('reviewers - new_reviewers', Array.from(new_reviewers).join(', '))
+
+  // Filter users to repository collaborators. If they're not, they can't be requested
+  // for review. In that case, they probably missed their invite to the maintainers team.
+  const reviewers = (
+    await Promise.all(
+      Array.from(new_reviewers, async (username) => {
+        try {
+          await github.rest.repos.checkCollaborator({
+            ...context.repo,
+            username,
+          })
```

Comment:
One thing that could happen on this endpoint is that the nixpkgs-ci app doesn't have the right privileges to check it. It used to be the codeowners app doing that.

We *should* have most of these privileges in nixpkgs-ci from https://github.com/NixOS/nixpkgs/pull/457566, but I'd still rather have an org owner available when merging this just in case.